shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java

```diff
@@ -190,6 +227,135 @@ public ASTNode visitCreateUser(final CreateUserContext ctx) {
+                if (null != foreignKeyOption) {
+                    result.getAllSQLSegments().add((TableSegment) visit(foreignKeyOption.referenceDefinition_().tableName()));
+                }
+                ChangeColumnSpecificationContext changeColumnSpecification = alterSpecification.changeColumnSpecification();
+                if (null != changeColumnSpecification) {
+                    createColumnPositionSegment(changeColumnSpecification.firstOrAfterColumn(),
+                            createColumnDefinitionSegment(changeColumnSpecification.columnDefinition(), result), result);
+                }
+                DropColumnSpecificationContext dropColumnSpecification = alterSpecification.dropColumnSpecification();
+                if (null != dropColumnSpecification) {
+                    result.getDroppedColumnNames().add(((ColumnSegment) visit(dropColumnSpecification)).getName());
+                }
+                ModifyColumnSpecificationContext modifyColumnSpecification = alterSpecification.modifyColumnSpecification();
+                if (null != modifyColumnSpecification) {
+                    createColumnPositionSegment(modifyColumnSpecification.firstOrAfterColumn(),
+                            createColumnDefinitionSegment(modifyColumnSpecification.columnDefinition(), result), result);
+                }
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public ASTNode visitDropTable(final DropTableContext ctx) {
+        DropTableStatement result = new DropTableStatement();
+        ListValue<TableSegment> tables = (ListValue<TableSegment>) visit(ctx.tableNames());
+        result.getTables().addAll(tables.getValues());
+        result.getAllSQLSegments().addAll(tables.getValues());
+        return result;
+    }
+
+    @Override
+    public ASTNode visitTruncateTable(final TruncateTableContext ctx) {
+        DDLStatement result = new DDLStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.getAllSQLSegments().add(table);
+        return result;
+    }
+
+    @Override
+    public ASTNode visitCreateIndex(final CreateIndexContext ctx) {
+        CreateIndexStatement result = new CreateIndexStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        IndexSegment index = (IndexSegment) visit(ctx.indexName());
+        result.setIndex(index);
+        result.getAllSQLSegments().add(index);
+        return result;
+    }
+
+    @Override
+    public ASTNode visitDropIndex(final DropIndexContext ctx) {
+        DropIndexStatement result = new DropIndexStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        IndexSegment index = (IndexSegment) visit(ctx.indexName());
```

Comment:
META-INF/parsing-rule-definition/mysql/sql-statement-rule-definition.xml tells us there is no need to add IndexSegment to CreateIndexStatement when DB is MySQL.

Reply:
@tristaZero Okay, I would like to remove `IndexSegment`.